File: pool_vd/src/core/models/PoolState/Ball.ts
import { Vector2i, IVector2i } from "./Vector2i";

export enum BallType {
  SOLID = "SOLID",
  STRIPED = "STRIPED"
}

export interface IBall {
  number: number;
  position: IVector2i;
  type: BallType;
  detectedAt: number;
}
export class Ball implements IBall {
  constructor(object: IBall) {
    this.number = object.number;
    this.type = object.type;
    this.position = new Vector2i(object.position.x, object.position.y);
    this.detectedAt = object.detectedAt;
  }
  number = 0;
  position: Vector2i = new Vector2i(0, 0);
  type: BallType = BallType.SOLID;
  detectedAt: number;
}

File: video_processor/src/pool_state/Ball.py
from enum import Enum
from .Vector2i import Vector2i

class BallType(Enum):
    SOLID = "SOLID"
    STRIPED = "STRIPED"

class Ball:
    def __init__(self, number: int, position: Vector2i, ball_type: BallType):
        self.number = number
        self.position = position
        self.type = ball_type